podcasts/Onboarding/Plus/SubscriptionPriceAndOfferView.swift

```diff
@@ -0,0 +1,89 @@
+import SwiftUI
+
+/// View to display the price of a subscription and any offer associated.
+struct SubscriptionPriceAndOfferView: View {
+
+    typealias ProductInfo = PlusPricingInfoModel.PlusProductPricingInfo
+
+    private let product: ProductInfo
+    private let mainTextColor: Color
+    private let secondaryTextColor: Color
+
+    init(product: ProductInfo, mainTextColor: Color, secondaryTextColor: Color) {
+        self.product = product
+        self.mainTextColor = mainTextColor
+        self.secondaryTextColor = secondaryTextColor
+    }
+
+    private func price(for subscriptionInfo: ProductInfo) -> AttributedString {
+        let subscriptionPeriod = subscriptionInfo.identifier.productInfo.frequency.description
+
+        guard let offer = subscriptionInfo.offer else {
+            var basePrice =  AttributedString(subscriptionInfo.rawPrice)
+            basePrice.font = .headline
+            basePrice.foregroundColor = mainTextColor
+
+            var basePeriod = AttributedString("/ \(subscriptionPeriod)")
+            basePeriod.foregroundColor = secondaryTextColor
+            basePeriod.font = .footnote
+
+            return basePrice + basePeriod
+        }
+
+
+        if offer.type == .freeTrial {
+            var basePrice =  AttributedString(subscriptionInfo.rawPrice)
+            basePrice.font = .headline
+            basePrice.foregroundColor = mainTextColor
+
+            var basePeriod = AttributedString("/\(subscriptionPeriod)")
+            basePeriod.foregroundColor = secondaryTextColor
+            basePeriod.font = .footnote
+
+            return basePrice + basePeriod
+        }
+
+        var offerPrice = AttributedString(offer.price)
+        offerPrice.foregroundColor = mainTextColor
+        offerPrice.font = .headline
+
+        var offerPeriod = AttributedString(" /\(subscriptionPeriod)")
+        offerPeriod.foregroundColor = secondaryTextColor
+        offerPeriod.font = .footnote
+
+        var basePrice = AttributedString(" \(offer.rawPrice)/\(subscriptionPeriod)")
+        basePrice.foregroundColor = secondaryTextColor
+        basePrice.font = .footnote
+        basePrice.strikethroughStyle = .single
+
+        return offerPrice + offerPeriod + basePrice
+    }
+
+    private func offerDescription(for subscriptionInfo: ProductInfo) -> String? {
+        guard let offer = subscriptionInfo.offer else {
+            return nil
+        }
+        return offer.description
+    }
+
+    var body: some View {
```

Comment:
What do you think about moving `body` to be the first property after the `init`?

I think it helps a lot because I can get a glimpse of what the view uses and then check the other methods.

Reply:
Will do